features/IFC102_Absence-of-deprecated-entities.feature

```diff
@@ -68,7 +74,80 @@ Background:
        | IfcGeographicElementType | "SOIL_BORING_POINT" |
 
 
+    Scenario Outline: Check for deprecated entities - IFC2X3
 
+      Given A model with Schema "IFC2X3"
+      Given An IFC model
 
+      Then There must be less than 1 instance(s) of <entity>
 
+      Examples:
+        | entity | 
+        | IfcConnectionPortGeometry | 
+        | Ifc2DCompositeCurve | 
+        | IfcElectricalElement | 
+        | IfcEquipmentElement | 
+
+
+    Scenario: Check for deprecated attributes - IFC2X3
+
+      Given A model with Schema "IFC2X3"
+      Given an IfcFillAreaStyleHatching
+
+      Then PointOfReferenceHatchLine = empty
+
+    
+    Scenario Outline: Check for deprecated explicitly instantiated entities - IFC2X3
+
+      Given A model with Schema "IFC2X3"
+      Given an <entity>
+
+      Then its type is not <entity> excluding subtypes
+
+    Examples:
+        | entity | 
+        | IfcProductRepresentation |
+        | IfcRepresentation | 
+        | IfcRepresentationContext | 
+        | IfcRelAssociates | 
+
+    Scenario: Check for deprecated property set - IFC2X3
+
+      Given a model with Schema "IFC2X3"
+      Given an IfcPropertySet
+
+      Then Name is not 'Pset_Draughting'
     
+
+    Scenario Outline: Check for deprecated entities - IFC4
+
+        Given A model with Schema "IFC4"
+        Given An IFC model
+
+        Then There must be less than 1 instance(s) of <entity>
+
+        Examples:
+          | entity                           | 
+          | IfcProxy                         | 
+          | IfcObjectTypeEnum                | 
+          | IfcOpeningStandardCase           | 
+          | IfcDoorStyle                     | 
+          | IfcWindowStyle                   | 
+          | IfcFaceBasedSurfaceModel         | 
+          | IfcMaterialClassificationRelationship | 
+          | IfcMaterialList                  | 
+          | IfcPresentationStyleAssignment   | 
+          | IfcNullStyle                     | 
+          | IfcPresentationStyleSelect       | 
+          | IfcStyleAssignmentSelect         | 
+          | IfcBeamStandardCase              | 
+          | IfcColumnStandardCase            | 
+          | IfcDoorStandardCase              | 
+          | IfcMemberStandardCase            | 
+          | IfcPlateStandardCase             | 
+          | IfcRelCoversSpaces               | 
+          | IfcSlabElementedCase             | 
+          | IfcSlabStandardCase              | 
+          | IfcWallElementedCase             | 
+          | IfcWallStandardCase              | 
+          | IfcWindowStandardCase            | 
```

Comment:
```suggestion
    | IfcWindowStandardCase            | 
        
Scenario Outline: Check for deprecated attributes - IFC4

      Given A model with Schema "IFC4"
      Given an <entity>

      Then <deprecated_attribute> = empty
      
      | examples |
      | entity | deprecated_attribute |
      | IfcRelAssignsToActor | RelatedObjectsType |
      | IfcRelAssignsToControl | RelatedObjectsType |
      | IfcRelAssignsToGroup | RelatedObjectsType |
      | IfcRelAssignsToGroupByFactor | RelatedObjects Type|
```

(you get the idea)

Reply:
Where did you find this list of deprecated attributes in IFC4? 
Or did you gather them manually?

Reply:
Added